diff -ur kazehakase-0.4.7.orig/etc/kzrc kazehakase-0.4.7.agent/etc/kzrc
--- kazehakase-0.4.7.orig/etc/kzrc	2007-07-13 19:18:59.000000000 +0200
+++ kazehakase-0.4.7.agent/etc/kzrc	2007-07-13 19:19:41.000000000 +0200
@@ -1,8 +1,6 @@
 # Encoding: UTF-8
 
 [Global]
-# Default layout engine. mozilla or gtk+-webcore
-#layout_engine=mozilla
 
 startup_page=about:blank
 
@@ -38,6 +36,10 @@
 # if javascript_enable is false in about:config.
 use_javascript=true
 
+[Browser]
+# Default layout engine. gecko or gtk+-webcore
+layout_engine=gecko
+
 [MainWindow]
 width=720
 height=600
diff -ur kazehakase-0.4.7.orig/module/embed/gecko/kz-gecko-single.cpp kazehakase-0.4.7.agent/module/embed/gecko/kz-gecko-single.cpp
--- kazehakase-0.4.7.orig/module/embed/gecko/kz-gecko-single.cpp	2007-07-13 19:19:00.000000000 +0200
+++ kazehakase-0.4.7.agent/module/embed/gecko/kz-gecko-single.cpp	2007-07-13 19:19:41.000000000 +0200
@@ -300,20 +300,6 @@
 }
 
 static void
-build_user_agent (gchar *user_agent, gint size)
-{
-	gchar *system;
-
-	system = kz_utils_get_system_name();
-
-	g_snprintf(user_agent, size,
-		   "Mozilla/5.0 (X11; %s; U;) Gecko/%d Kazehakase/"  VERSION,
-		   system,
-		   NS_BUILD_ID/100);
-	g_free (system);
-}
-
-static void
 set_user_agent (KzProfile *profile)
 {
 	gchar tmp_string[1024];
@@ -326,12 +312,16 @@
 		value = kz_profile_get_value(profile, "Global", "user_agent", 
 				&tmp_string, G_N_ELEMENTS(tmp_string),
 				KZ_PROFILE_VALUE_TYPE_STRING);
+		mozilla_prefs_set_string("general.useragent.override", tmp_string);
 	}
 	else /* set default */
 	{
-		build_user_agent(tmp_string, G_N_ELEMENTS(tmp_string));
+		mozilla_prefs_delete ("general.useragent.override");
+		mozilla_prefs_set_string ("general.useragent.extra.kazehakase", "Kazehakase/" VERSION);
+		mozilla_prefs_set_string ("general.useragent.vendor", "PLD");
+		mozilla_prefs_set_string ("general.useragent.vendorSub", "2.0");
+		mozilla_prefs_set_string ("general.useragent.vendorComment", "Ac");
 	}
-	mozilla_prefs_set_string("general.useragent.override", tmp_string);
 }
 
 static void
diff -ur kazehakase-0.4.7.orig/module/embed/gecko/mozilla-prefs.cpp kazehakase-0.4.7.agent/module/embed/gecko/mozilla-prefs.cpp
--- kazehakase-0.4.7.orig/module/embed/gecko/mozilla-prefs.cpp	2007-07-13 19:19:00.000000000 +0200
+++ kazehakase-0.4.7.agent/module/embed/gecko/mozilla-prefs.cpp	2007-07-13 19:19:41.000000000 +0200
@@ -449,6 +449,26 @@
 }
 
 
+gboolean
+mozilla_prefs_delete (const char *preference_name)
+{
+	g_return_val_if_fail(preference_name != NULL, FALSE);
+
+	nsCOMPtr<nsIPrefService> prefService = 
+				do_GetService(NS_PREFSERVICE_CONTRACTID);
+	nsCOMPtr<nsIPrefBranch> pref;
+	prefService->GetBranch("", getter_AddRefs(pref));
+
+	if (pref)
+	{
+		nsresult rv = pref->ClearUserPref(preference_name);
+		return NS_SUCCEEDED(rv) ? TRUE : FALSE;
+	}
+
+	return FALSE;
+}
+
+
 void
 mozilla_prefs_set_use_proxy (gboolean use)
 {
diff -ur kazehakase-0.4.7.orig/module/embed/gecko/mozilla-prefs.h kazehakase-0.4.7.agent/module/embed/gecko/mozilla-prefs.h
--- kazehakase-0.4.7.orig/module/embed/gecko/mozilla-prefs.h	2007-07-13 19:19:00.000000000 +0200
+++ kazehakase-0.4.7.agent/module/embed/gecko/mozilla-prefs.h	2007-07-13 19:19:41.000000000 +0200
@@ -46,6 +46,7 @@
 					gboolean     new_boolean_value);
 gboolean mozilla_prefs_set_int         (const char  *preference_name,
 					int          new_int_value);
+gboolean mozilla_prefs_delete          (const char  *preference_name);
 void     mozilla_prefs_set_proxy       (KzProxyItem *item);
 void     mozilla_prefs_set_use_proxy   (gboolean use);
 
