Index: trunk/src/kz-notebook.c
===================================================================
--- trunk/src/kz-notebook.c	(revision 3163)
+++ trunk/src/kz-notebook.c	(working copy)
@@ -424,12 +424,9 @@
 	/* add to this notebook's history */
 	priv->open_hist = g_list_prepend(priv->open_hist, embed);
 
-	if(pos >= 0)
-	{
-		sibembed = gtk_notebook_get_nth_page(GTK_NOTEBOOK(notebook), pos);
-		sibtab = KZ_TAB_LABEL(gtk_notebook_get_tab_label(GTK_NOTEBOOK(notebook),
-				      sibembed));
-	}
+	sibembed = gtk_notebook_get_nth_page(GTK_NOTEBOOK(notebook), pos);
+	sibtab = KZ_TAB_LABEL(gtk_notebook_get_tab_label(GTK_NOTEBOOK(notebook),
+				sibembed));
 	return sibtab;
 }
 
