diff -ur kazehakase-0.4.4.1.orig/src/mozilla/mozilla-prefs.cpp kazehakase-0.4.4.1/src/mozilla/mozilla-prefs.cpp
--- kazehakase-0.4.4.1.orig/src/mozilla/mozilla-prefs.cpp	2007-01-02 10:23:47.000000000 +0100
+++ kazehakase-0.4.4.1/src/mozilla/mozilla-prefs.cpp	2007-03-22 00:38:00.000000000 +0100
@@ -62,31 +62,6 @@
 }
 
 
-static void
-build_user_agent (gchar *user_agent, gint size)
-{
-	struct utsname name;
-	gchar *system;
-
-	if (uname (&name) >= 0)
-	{
-		system = g_strdup_printf ("%s %s",
-					  name.sysname, 
-					  name.machine);
-	}
-	else
-	{
-		system = g_strdup ("Unknown");
-	}
-		
-	g_snprintf(user_agent, size,
-		   "Mozilla/5.0 (X11; %s; U;) Gecko/%d Kazehakase/"  VERSION,
-		   system,
-		   NS_BUILD_ID/100);
-	g_free (system);
-}
-
-
 extern "C" void 
 mozilla_prefs_set (KzProfile *profile)
 {
@@ -106,11 +81,16 @@
 		override = kz_profile_get_value(profile, "Global", "user_agent", 
 			   			&user_agent, sizeof(user_agent),
 						KZ_PROFILE_VALUE_TYPE_STRING);
+		mozilla_prefs_set_string ("general.useragent.override", user_agent);
 	}
 	if (!override)
-		build_user_agent(user_agent, G_N_ELEMENTS(user_agent));
-	mozilla_prefs_set_string ("general.useragent.override", user_agent);
-
+	{
+		mozilla_prefs_delete ("general.useragent.override");
+		mozilla_prefs_set_string ("general.useragent.extra.kazehakase", "Kazehakase/" VERSION);
+		mozilla_prefs_set_string ("general.useragent.vendor", "PLD");
+		mozilla_prefs_set_string ("general.useragent.vendorSub", "2.0");
+		mozilla_prefs_set_string ("general.useragent.vendorComment", "Ac");
+	}
 	if (!mozilla_prefs_get_string("kazehakase.font.language", &value))
 	{
 		mozilla_prefs_set_string("kazehakase.font.language",
@@ -308,6 +288,26 @@
 }
 
 
+extern "C" gboolean
+mozilla_prefs_delete(const char *preference_name)
+{
+	g_return_val_if_fail(preference_name != NULL, FALSE);
+
+	nsCOMPtr<nsIPrefService> prefService = 
+				do_GetService (NS_PREFSERVICE_CONTRACTID);
+	nsCOMPtr<nsIPrefBranch> pref;
+	prefService->GetBranch("", getter_AddRefs(pref));
+
+	if (pref)
+	{
+		nsresult rv = pref->ClearUserPref(preference_name);
+		return NS_SUCCEEDED(rv) ? TRUE : FALSE;
+	}
+
+	return FALSE;
+}
+
+
 //
 //  Picked from Epiphany-0.7.0 (embed/mozilla/mozilla-embed-single.cpp).
 //  Copyright (C) 2000, 2001, 2002 Marco Pesenti Gritti
diff -ur kazehakase-0.4.4.1.orig/src/mozilla/mozilla-prefs.h kazehakase-0.4.4.1/src/mozilla/mozilla-prefs.h
--- kazehakase-0.4.4.1.orig/src/mozilla/mozilla-prefs.h	2007-01-02 10:23:47.000000000 +0100
+++ kazehakase-0.4.4.1/src/mozilla/mozilla-prefs.h	2007-03-22 00:37:24.000000000 +0100
@@ -46,6 +46,7 @@
 					gboolean     new_boolean_value);
 gboolean mozilla_prefs_set_int         (const char  *preference_name,
 					int          new_int_value);
+gboolean mozilla_prefs_delete          (const char  *preference_name);
 
 gboolean mozilla_prefs_get_font_list   (const char  *lang_group,
 					const char  *font_type,
