diff -urN kazehakase-0.2.1.org/data/kazehakase.desktop kazehakase-0.2.1/data/kazehakase.desktop
--- kazehakase-0.2.1.org/data/kazehakase.desktop	2004-11-04 20:27:36.972728128 +0100
+++ kazehakase-0.2.1/data/kazehakase.desktop	2004-11-04 20:37:51.302335832 +0100
@@ -2,11 +2,13 @@
 Encoding=UTF-8
 Name=Kazehakase Web Browser
 Name[ja]=風博士ウェブブラウザ
+Name[pl]=Przeglądarka Kazehakase
 Comment=Web browser
 Comment[ja]=ウェブブラウザ
+Comment[pl]=Przeglądarka WWW
 TryExec=kazehakase
 Exec=kazehakase %U
 Terminal=false
 Type=Application
 Icon=kazehakase-icon.png
-Categories=Application;Network;
+Categories=GTK;Network;WebBrowser;
