diff -urN kazehakase-0.4.8/src/actions/Makefile.am kazehakase-0.4.8.new/src/actions/Makefile.am
--- kazehakase-0.4.8/src/actions/Makefile.am	2007-08-29 17:45:09.000000000 +0200
+++ kazehakase-0.4.8.new/src/actions/Makefile.am	2007-09-22 12:12:37.000000000 +0200
@@ -6,7 +6,6 @@
 CLEANFILES =
 
 AM_CPPFLAGS =					\
-	-DGTK_DISABLE_DEPRECATED=1		\
 	-DGDK_DISABLE_DEPRECATED=1		\
 	-DG_LOG_DOMAIN=\"Kazehakase-Actions\"	\
 	-DG_DISABLE_DEPRECATED=1
diff -urN kazehakase-0.4.8/src/bookmarks/Makefile.am kazehakase-0.4.8.new/src/bookmarks/Makefile.am
--- kazehakase-0.4.8/src/bookmarks/Makefile.am	2007-08-29 17:45:09.000000000 +0200
+++ kazehakase-0.4.8.new/src/bookmarks/Makefile.am	2007-09-22 12:12:28.000000000 +0200
@@ -14,7 +14,6 @@
 	-I$(top_srcdir)/src
 
 AM_CPPFLAGS= \
-	-DGTK_DISABLE_DEPRECATED=1 \
 	-DGDK_DISABLE_DEPRECATED=1 \
 	-DG_LOG_DOMAIN=\"Kazehakase-Bookmarks\" \
 	-DG_DISABLE_DEPRECATED=1 \
diff -urN kazehakase-0.4.8/src/dialogs/Makefile.am kazehakase-0.4.8.new/src/dialogs/Makefile.am
--- kazehakase-0.4.8/src/dialogs/Makefile.am	2007-08-29 17:45:08.000000000 +0200
+++ kazehakase-0.4.8.new/src/dialogs/Makefile.am	2007-09-22 12:12:47.000000000 +0200
@@ -14,7 +14,6 @@
 
 AM_CPPFLAGS= \
 	-DKZ_DATADIR=\""$(datadir)/$(PACKAGE)"\" \
-	-DGTK_DISABLE_DEPRECATED=1 \
 	-DGDK_DISABLE_DEPRECATED=1 \
 	-DG_LOG_DOMAIN=\"Kazehakase-Dialogs\" \
 	-DG_DISABLE_DEPRECATED=1
diff -urN kazehakase-0.4.8/src/Makefile.am kazehakase-0.4.8.new/src/Makefile.am
--- kazehakase-0.4.8/src/Makefile.am	2007-08-29 17:45:10.000000000 +0200
+++ kazehakase-0.4.8.new/src/Makefile.am	2007-09-22 12:14:00.000000000 +0200
@@ -18,7 +18,6 @@
 	-DKZ_DATADIR=\""$(datadir)/$(PACKAGE)"\"	\
 	-DKZ_SEARCH_MODULEDIR=\""$(searchdir)"\"	\
 	-DKZ_EMBED_MODULEDIR=\""$(embeddir)"\"		\
-	-DGTK_DISABLE_DEPRECATED=1			\
 	-DGDK_DISABLE_DEPRECATED=1			\
 	-DG_LOG_DOMAIN=\"Kazehakase\"			\
 	-DG_DISABLE_DEPRECATED=1
diff -urN kazehakase-0.4.8/src/net/Makefile.am kazehakase-0.4.8.new/src/net/Makefile.am
--- kazehakase-0.4.8/src/net/Makefile.am	2007-08-29 17:45:09.000000000 +0200
+++ kazehakase-0.4.8.new/src/net/Makefile.am	2007-09-22 12:13:16.000000000 +0200
@@ -15,7 +15,6 @@
 	$(LIBGNUTLS_CFLAGS)
 
 AM_CPPFLAGS= \
-	-DGTK_DISABLE_DEPRECATED=1 \
 	-DGDK_DISABLE_DEPRECATED=1 \
 	-DG_LOG_DOMAIN=\"Kazehakase-Net\" \
 	-DG_DISABLE_DEPRECATED=1
diff -urN kazehakase-0.4.8/src/prefs_ui/Makefile.am kazehakase-0.4.8.new/src/prefs_ui/Makefile.am
--- kazehakase-0.4.8/src/prefs_ui/Makefile.am	2007-08-29 17:45:10.000000000 +0200
+++ kazehakase-0.4.8.new/src/prefs_ui/Makefile.am	2007-09-22 12:13:24.000000000 +0200
@@ -11,7 +11,6 @@
 	-I$(top_srcdir)/src
 
 AM_CPPFLAGS =					\
-	-DGTK_DISABLE_DEPRECATED=1		\
 	-DGDK_DISABLE_DEPRECATED=1		\
 	-DG_LOG_DOMAIN=\"Kazehakase-PrefsUI\"	\
 	-DG_DISABLE_DEPRECATED=1
diff -urN kazehakase-0.4.8/src/sidebar/Makefile.am kazehakase-0.4.8.new/src/sidebar/Makefile.am
--- kazehakase-0.4.8/src/sidebar/Makefile.am	2007-08-29 17:45:09.000000000 +0200
+++ kazehakase-0.4.8.new/src/sidebar/Makefile.am	2007-09-22 12:12:06.000000000 +0200
@@ -14,7 +14,6 @@
 	-I$(top_srcdir)/src
 
 AM_CPPFLAGS= \
-	-DGTK_DISABLE_DEPRECATED=1 \
 	-DGDK_DISABLE_DEPRECATED=1 \
 	-DG_LOG_DOMAIN=\"Kazehakase-Sidebar\" \
 	-DG_DISABLE_DEPRECATED=1
diff -urN kazehakase-0.4.8/src/utils/Makefile.am kazehakase-0.4.8.new/src/utils/Makefile.am
--- kazehakase-0.4.8/src/utils/Makefile.am	2007-08-29 17:45:09.000000000 +0200
+++ kazehakase-0.4.8.new/src/utils/Makefile.am	2007-09-22 12:13:33.000000000 +0200
@@ -16,7 +16,6 @@
 	-I$(top_srcdir)/src/libegg/pixbufthumbnail
 
 AM_CPPFLAGS= \
-	-DGTK_DISABLE_DEPRECATED=1 \
 	-DGDK_DISABLE_DEPRECATED=1 \
 	-DG_LOG_DOMAIN=\"Kazehakase-Utils\" \
 	-DG_DISABLE_DEPRECATED=1
