--- kazehakase-0.1.6/src/libegg/egg-toolbar.h.oryg	2004-06-22 13:20:28.397406168 +0200
+++ kazehakase-0.1.6/src/libegg/egg-toolbar.h	2004-06-22 13:27:20.652733832 +0200
@@ -189,6 +189,7 @@
 void       egg_toolbar_remove_space    (EggToolbar      *toolbar,
                                         gint             position);
 /* Any element type */
+/*
 GtkWidget* egg_toolbar_append_element  (EggToolbar      *toolbar,
 					EggToolbarChildType type,
 					GtkWidget       *widget,
@@ -208,7 +209,7 @@
 					GtkWidget       *icon,
 					GtkSignalFunc    callback,
 					gpointer         user_data);
-
+*/
 GtkWidget* egg_toolbar_insert_element  (EggToolbar      *toolbar,
 					EggToolbarChildType type,
 					GtkWidget       *widget,
