diff -ur kazehakase-0.1.9.orig/data/kazehakase.desktop kazehakase-0.1.9/data/kazehakase.desktop
--- kazehakase-0.1.9.orig/data/kazehakase.desktop	2004-05-08 06:23:18.000000000 +0200
+++ kazehakase-0.1.9/data/kazehakase.desktop	2004-09-24 21:25:51.725344512 +0200
@@ -1,12 +1,15 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Kazehakase Web Browser
+Name[pl]=Przeglądarka Kazehakase
 Name[ja]=風博士ウェブブラウザ
 Comment=Web browser
+Comment=Przeglądarka WWW
 Comment[ja]=ウェブブラウザ
 TryExec=kazehakase
 Exec=kazehakase %U
 Terminal=false
 Type=Application
 Icon=kazehakase-icon.png
-Categories=Application;Network;
+Categories=GTK;Network;WebBrowser;
+# vi: encoding=utf-8
